packages/core/sdk/__tests__/auth/internal/CoreStitchAuthUnitTests.ts

```diff
@@ -248,46 +253,285 @@ describe("CoreStitchAuthUnitTests", () => {
+
+    mockNextUserResponse(requestClientMock);
+    const user1 = await auth.loginWithCredentialInternal(
+      new AnonymousCredential()
+    );
+
+    mockNextUserResponse(requestClientMock);
+    const user2 = await auth.loginWithCredentialInternal(
+      new UserPasswordCredential("hi", "there")
+    );
+
+    mockNextUserResponse(requestClientMock);
+    const user3 = await auth.loginWithCredentialInternal(
+      new UserPasswordCredential("bye", "there")
+    );
+
+    expect(auth.listUsers()[2]).toEqual(user3);
+    expect(auth.isLoggedIn).toBeTruthy();
+
+    await auth.removeUserInternal();
+
+    // assert that though one user was removed and we are logged out, 
+    // two users are still listed.
+    expect(auth.isLoggedIn).toBeFalsy();
+    expect(auth.listUsers().length).toEqual(2);
+
+    verify(requestClientMock.doRequest(anything())).times(7);
+
+    const expectedRequest = new StitchRequest.Builder();
+    expectedRequest.withMethod(Method.DELETE).withPath(routes.sessionRoute);
+    const headers = {
+      [Headers.AUTHORIZATION]: Headers.getAuthorizationBearer(
+        TEST_REFRESH_TOKEN
+      )
+    };
+    expectedRequest.withHeaders(headers);
+
+    let [actualRequest] = capture(
+      requestClientMock.doRequest
+    ).byCallIndex(6);
+    expect(expectedRequest.build()).toEqualRequest(actualRequest);
+
+    expect(auth.isLoggedIn).toBeFalsy();
+
+    // assert that switching to a user, and removing self works
+    auth.switchToUserWithId(user2.id);
+    expect(auth.isLoggedIn).toBeTruthy();
+
+    await auth.removeUserInternal();
+
+    verify(requestClientMock.doRequest(anything())).times(8);
+
+    [actualRequest] = capture(
+      requestClientMock.doRequest
+    ).byCallIndex(7);
+    expect(expectedRequest.build()).toEqualRequest(actualRequest);
+
+    // assert that there is one user left
+    expect(auth.listUsers().length).toEqual(1);
+    expect(auth.listUsers()[0]).toEqual(user1);
+
+    // assert that we can remove the user without switching to it
+    await auth.removeUserWithIdInternal(user1.id);
+    expect(auth.listUsers().length).toEqual(0);
+    expect(auth.isLoggedIn).toBeFalsy();
+
+    // assert that we can log back into the user that we removed
+    mockNextUserResponse(requestClientMock, user2.id);
+    expect(await auth.loginWithCredentialInternal(
+      new UserPasswordCredential("hi", "there")
+    )).toEqual(user2);
+
+    expect(auth.listUsers().length).toEqual(1);
+    expect(auth.listUsers()[0]).toEqual(user2);
+
+    expect(auth.isLoggedIn).toBeTruthy();
+  });
+
+  it("should be able to switch users", async () => {
+    const requestClientMock = getMockedRequestClient();
+    const requestClient = instance(requestClientMock);
+    const routes = new StitchAppRoutes("my_app-12345").authRoutes;
+    const storage = new MemoryStorage(appId);
+    const auth = new StitchAuth(
+      requestClient,
+      routes,
+      storage
+    );
+
+    try {
```

Comment:
re my comment above, I see that you handle failures here, but may still be nice to do it for all of the methods that throw